usermods/rc_switch_v2/rc_switch_v2.h

```diff
@@ -0,0 +1,72 @@
+#pragma once
+
+#include "wled.h"
+
+#include <RCSwitch.h>
+
+#ifndef USERMOD_RC_SWITCH_PIN
+#ifdef ARDUINO_ARCH_ESP32
+#define USERMOD_RC_SWITCH_PIN 22
+#else //ESP8266 boards
+#define USERMOD_RC_SWITCH_PIN 4
+#endif
+#endif
+
+#ifndef USERMOD_RC_SWITCH_PULSE_LENGTH
+#define USERMOD_RC_SWITCH_PULSE_LENGTH 50
+#endif
+
+#ifndef USERMOD_RC_SWITCH_PROTOCOL
+#define USERMOD_RC_SWITCH_PROTOCOL 15
+#endif
+
+RCSwitch mySwitch = RCSwitch();
+
+class RcSwitchV2 : public Usermod {
+
+  private:
+
+    bool initDone = false;
+    static const char _name[];
+
+  public:
+
+    void setup() {
+      mySwitch.enableTransmit(USERMOD_RC_SWITCH_PIN);
+      mySwitch.setPulseLength(USERMOD_RC_SWITCH_PULSE_LENGTH);
+      mySwitch.setProtocol(USERMOD_RC_SWITCH_PROTOCOL);
+
+      initDone = true;
+    }
+    
+    void loop() {}
+
+    void readFromJsonState(JsonObject& root)
+    {
+      if (!initDone) return;  // prevent crash on boot applyPreset()
+
+      JsonObject usermod = root[FPSTR(_name)];
+      if (!usermod.isNull()) {
+        mySwitch.send(usermod["rcswitch"]);
```

Comment:
Also not sure if json `usermod["rcswitch"]` will be available and will it be a `char *` type as needed?

Reply:
Cast it using `.to<const char *>()` method. It will return null pointer if it is not string.

Reply:
how exactly do I cast it?

`const char * test = usermod["rcswitch"].to<const char *>()`

I am getting error:

error: no matching function for call to 'ArduinoJson6181_90::MemberProxy<ArduinoJson6181_90::ObjectRef, const char*>::to()

Reply:
Ah. sorry. It is `.as<const char *>()`.
Please consult ArduinoJson documentation.